File: testprojects/scala_main/pipelines/SCALA_DEP_MGMT_ALL/code/src/main/scala/org/main/scla_dep_mgmt/udfs/UDFs.scala
package org.main.scla_dep_mgmt.udfs

import _root_.io.prophecy.abinitio.ScalaFunctions._
import _root_.io.prophecy.libs._
import org.apache.spark.sql.types._
import org.apache.spark.sql.functions._
import org.apache.spark.sql._

object UDFs extends Serializable {
  var int_value    = 10
  var string_value = "string value"

  def registerUDFs(spark: SparkSession) = {
    spark.udf.register("udf_multiply",         udf_multiply)
    spark.udf.register("udf_string_null_safe", udf_string_null_safe)
    spark.udf.register("udf_string_length",    udf_string_length)
  }

  def udf_multiply = udf((value: Int) => value * int_value)

  def udf_string_null_safe =
    udf((s: String) => if (s != null) s.length else string_value.length)

  def udf_string_length = udf((s: String) => s.length)
}

File: testprojects/python_main/pipelines/IO_PYTHON_BASIC/code/pythonbasic/test/mainone/udfs/UDFs.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.lookups import (
    createLookup,
    createRangeLookup,
    lookup,
    lookup_last,
    lookup_match,
    lookup_count,
    lookup_row,
    lookup_row_reverse,
    lookup_nth
)
int_value = 15
N = 10

def registerUDFs(spark: SparkSession):
    spark.udf.register("squared_udf", squared_udf)

@udf(returnType = IntegerType())
def squared_udf(value=10):
    return ((value * value) + int_value - float_value) if value else int_value
